### ISSUE TITLE
Remove ClassificationBundle services from backend

### ISSUE THREAD

**Issue by clytemnestra** [sonata-project/SonataMediaBundle#917]
By default, ClassificationBundle adds in backend admin management for categories, tags, collections and contexts, but since my application doesn't use them I want to remove them from the menu and admin panel.

I've never removed a service before, so I don't know how to do it.

There has to be a way to remove these services from SonataClassificationBundle/Resources/config/admin.xml, obviously without modifying the file itself because it is a vendor file.

Or maybe is there a way to remove them from the Sonata admin pool? Since they are tagged with sonata.admin?

**Comment by clytemnestra** [comment]
So I managed to do it by using a compiler pass in a Sonata Admin Bundle extension using Easy Extends.

```
class CustomCompilerPass implements CompilerPassInterface
{
    public function process(ContainerBuilder $container)
    {
        $container->removeDefinition('sonata.classification.admin.category');

        $container->removeDefinition('sonata.classification.admin.tag');

        $container->removeDefinition('sonata.classification.admin.collection');

        $container->removeDefinition('sonata.classification.admin.context');
    }
}
```

However, when I create a new Gallery, and add a new Media to it and try to create a new image and add it, I get 

```
Impossible to invoke a method ("id") on a null variable in SonataDoctrineORMAdminBundle:Form:form_admin_fields.html.twig at line 59






in app\cache\dev\classes.php at line 5005   -
} else {
$message = sprintf('Impossible to invoke a method ("%s") on a %s variable ("%s")', $item, gettype($object), $object);
}
throw new Twig_Error_Runtime($message, -1, $this->getTemplateName());
}
if (self::METHOD_CALL !== $type && !$object instanceof self) { if (isset($object->$item) || array_key_exists((string) $item, $object)) {
if ($isDefinedTest) {
```

So I guess the services are needed somewhere in there. I guess we'll have to wait for https://github.com/sonata-project/SonataMediaBundle/pull/794

**Comment by clytemnestra** [comment]
Half a year later, I found a much cleaner way that keeps the services in the container(because they are needed somewhere), but doesn't show them on the admin panel.

```
       $definitionsNames = array('sonata.media.admin.media', 'sonata.media.admin.gallery_has_media', 'sonata.media.admin.gallery',
            'sonata.classification.admin.category','sonata.classification.admin.tag','sonata.classification.admin.context','sonata.classification.admin.collection');

        foreach ($definitionsNames as $definitionName) {
            $definition = $container->getDefinition($definitionName);

            $tags = $definition->getTags();

            $tags['sonata.admin'][0]['show_in_dashboard'] = false;
            $definition->setTags($tags);

        }
```

Unfortunately, the admin routes are still available. Not an issue for my side, but I believe there are ways to remove them.
The thing is the media_widget contains a link to the admin media edit route, so that needs to be overwritten to not have it anymore. Then the Media, Gallery and GHM admins need to be overridden, and override the configureroutes() function and remove all routes. Then I believe you can't access anything through the admin, but the application can still use the admin services, if they are needed anywhere.

**Comment by jordisala1991** [comment]
Fixed on latest release of MediaBundle 3.4.0. ClassificationBundle is no longer a required dependency
